pkg/jobs/hive/hive.go

```diff
@@ -1011,7 +1013,7 @@ func eusRetreiveAndUpdateClusterVersion(
 	if err := client.Get(context.TODO(), types.NamespacedName{
 		Namespace: clusterName,
 		Name:      clusterName,
-	}, &mcview); err != nil {
+	}, &mcview); err != nil && k8serrors.IsNotFound(err) {
```

Comment:
you should do something like this:
if err exists and is a notFound err then create, if it's not a notFound err then return the err.
same comment for the other occurrences. sry I wasn't more clear before.